languages/ruby/exercises/concept/instance-variables/.docs/hints.md

```diff
@@ -0,0 +1,39 @@
+## General
+
+About initializing object instances:
+
+- [Ruby Guides: Initialize Method][rg-initialize-method]
+
+About instance variables:
+
+- [Ruby For Beginners: Instance variables][rfb-instance-variables]
+- [Ruby Guides: Instance variables][rg-instance-variables]
+- [Ruby User's Guide: Instance variables][rug-instance-variables]
+- [Geeks for Geeks: Ruby Getters and Setters Methods][gfg-getter-setters]
+
+## 1. Make new attendees
+
+- Using the references, complete the initialize method to save the height to the object's state.
+
+## 2. How tall is the attendee
+
+- Create a getter for the saved height state.
+
+## 3. What is the ride pass's id
+
+- Create a getter for the pass id.
+- It is okay and expected to return `nil` before a pass is issued to the `Attendee`.
+
+## 4. Allow people to buy a pass
+
+- Using the `issue_pass!` setter method set the instance's state to the argument.
```

Comment:
The "bang" is OK here, `issue_pass` implies that state is being changed.  And there isn't a "nondestructive" method of the same name.  Ruby convention has been that if there is a non-destructive and a destructive (or modifying) version as well, to use the bang, otherwise, the name should indicate this well enough if there is a change of the object being done there.

> Examples of such pairs of methods include sort/sort! for arrays, upcase/upcase! for strings, chomp/chomp! for strings, and reverse/reverse! for strings and arrays. In each case, if you call the non-bang version of the method on the object, you get a new object. If you call the bang version, you operate in-place on the same object to which you sent the message. ¹

---

[1] Writing Own Ruby Methods: Ruby Study Notes - Best Ruby Guide, Ruby Tutorial, (n.d.). http://rubylearning.com/satishtalim/writing_own_ruby_methods.html (accessed July 18, 2020).

Reply:
I would lean toward not having the bang though. (I forgot to mention that).